### PR TITLE
Read receipts are saved to the local database with the same primary key

### PR DESCRIPTION
**Pull request by dogacsenol**
Read receipts are saved to the local database with the same primary key (_id = 0). It is OK if you don't reinstall the app, as there is another column for messages in the database for "read" flag, but when reinstalling the app, the messages are redownloaded from Firebase and the read flags are not set correctly.

This is because the CoreEntity for ReadReceiptUserLink is created with a primary key of "long" (primitive) not "Long" (class). 

When it is created with "long" instead of "Long", greenDAO generates the Dao DDL with "\"_id\" INTEGER PRIMARY KEY NOT NULL. I am no SQLite expert but have lots of experience with enterprise level databases, in which normally NOT NULL makes sense for a PRIMARY KEY.

I don't know if this is a bug (or feature) of SQLite or greendao, but when _id is created with NOT NULL statement in SQLite, it is not mapped to the internal autoincrement ROWID, therefore always stays 0. All read receipts are inserted into the same row with the id 0. I checked this (after late night hair pulling for a couple of hours) by injecting a counter into readReceiptId and afterwards downloading the sqlite database to my computer.

Long story short, changing the primitive long to class Long, the table is generated correctly.

I also had to set readReceiptId to the Message's Id manually. I would expect an automatic assignment from greendao, but it did not work that way.

Now, I can remove the app and install again, the "read" status of messages are persisted.

### CODE REVIEW

File: chat-sdk-core/src/main/java/co/chatsdk/core/dao/Message.java [hunk 1]
```diff
@@ -254,6 +254,7 @@ public void setUserReadStatus (User user, ReadStatus status, DateTime date) {
         ReadReceiptUserLink link = linkForUser(user);
         if(link == null) {
             link = StorageManager.shared().createEntity(ReadReceiptUserLink.class);
+            link.setReadReceiptId(this.getId());
```
Comment (bensmiley) [inline chat-sdk-core/src/main/java/co/chatsdk/core/dao/Message.java]: Do you know if this step is necessary? Is it possible that the `readReceiptId` and `userId` need to be set to `Long` too?

Comment (dogacsenol) [reply chat-sdk-core/src/main/java/co/chatsdk/core/dao/Message.java]: I was expecting greenDAO to resolve the foreign key automatically, but it did not. I also did not have much of a time to see why, so I just assigned it manually.

For the second question, yes I believe so.
